sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -0,0 +1,895 @@
+    This allows HDBSCAN to find clusters of varying densities (unlike DBSCAN),
+    and be more robust to parameter selection.
+
+    Parameters
+    ----------
+    min_cluster_size : int, default=5
+        The minimum number of samples in a group for that group to be
+        considered a cluster; groupings smaller than this size will be left
+        as noise.
+
+    min_samples : int, default=None
+        The number of samples in a neighborhood for a point
+        to be considered as a core point. This includes the point itself.
+        defaults to the `min_cluster_size`.
+
+    cluster_selection_epsilon : float, default=0.0
+        A distance threshold. Clusters below this value will be merged.
+        See [5]_ for more information.
+
+    max_cluster_size : int, default=0
+        A limit to the size of clusters returned by the `eom` cluster selection
+        algorithm. Has no effect if `cluster_selection_method=leaf`. Can be
+        overridden in rare cases by a high value for
+        `cluster_selection_epsilon`.
+
+    metric : str or callable, default='euclidean'
+        The metric to use when calculating distance between instances in a
+        feature array.
+
+        - If metric is a string or callable, it must be one of
+          the options allowed by `metrics.pairwise.pairwise_distances` for its
+          metric parameter.
+
+        - If metric is "precomputed", X is assumed to be a distance matrix and
+          must be square.
+
+    alpha : float, default=1.0
+        A distance scaling parameter as used in robust single linkage.
+        See [3]_ for more information.
+
+    algorithm : str, default='auto'
+        Exactly which algorithm to use; hdbscan has variants specialised
+        for different characteristics of the data. By default this is set
+        to `'auto'` which attempts to use a `KDTree` tree if possible,
+        otherwise it uses a `BallTree` tree.
+
+        If the `X` passed during `fit` is sparse or `metric` is invalid for
+        both `KDTree` and `BallTree`, then it resolves to use the `brute`
+        algorithm.
+
+        Available algorithms:
+        - `'brute'`
+        - `'kdtree'`
+        - `'balltree'`
+
+    leaf_size : int, default=40
+        Leaf size for trees responsible for fast nearest neighbour queries. A
+        large dataset size and small leaf_size may induce excessive memory
+        usage. If you are running out of memory consider increasing the
+        `leaf_size` parameter. Ignored for `algorithm=brute`.
+
+    memory : str, default=None
+        Used to cache the output of the computation of the tree.
+        By default, no caching is done. If a string is given, it is the
+        path to the caching directory.
+
+    n_jobs : int, default=4
+        Number of parallel jobs to run in core distance computations (if
+        supported by the specific algorithm). For `n_jobs<0`,
+        `(n_cpus + n_jobs + 1)` are used.
```

Comment:
We can use the same description we use everywhere for `n_jobs`:

https://github.com/scikit-learn/scikit-learn/blob/dff6081a915d9e4b8841ddb2a58d433e51883f1a/sklearn/compose/_column_transformer.py#L101-L103